crates/flox-rust-sdk/src/models/publish.rs

```diff
@@ -203,6 +204,37 @@ impl<'flox> Publish<'flox, Empty> {
 }
 
 impl<'flox> Publish<'flox, NixAnalysis> {
+    /// Sign the binary
+    ///
+    /// Requires a valid signing key
+    pub async fn sign_binary(
+        self,
+        key_file: impl AsRef<Path>,
+    ) -> Result<Publish<'flox, NixAnalysis>, PublishError> {
```

Comment:
I don't know if we concluded to _require_ signing or whether that's optional.
If it's required we should encode that in the type-state and transition with this call.
The current impl makes signing optional!

Reply:
Since we haven't figured out an easy way to do key management I'd expect it to stay optional for now